### PR TITLE
IGNITE-15694 Document new building process of Ignite C++ and ODBC on Windows

### PR DESCRIPTION
**Pull request by ivandasch**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: docs/_docs/quick-start/cpp.adoc [hunk 1]
```diff
@@ -32,24 +32,48 @@ include::includes/install-ignite.adoc[]
...
-Because this guide focuses on the thin client, you can run the examples below, connecting to the Java-based nodes you just started.
+Because this guide focuses on the thick client, you can run the examples below, connecting to the Java-based nodes you just started.
 
-Once the cluster is started, you can use the Ignite C++ thin client to perform cache operations (things like getting or putting data, or using SQL).
+Once the cluster is started, you can use the Ignite C++ thick client to perform cache operations (things like getting or putting data, or using SQL).
 
 == Getting Started with Ignite and C++
 
-Ignite ships with a robust {cpp} client.
+Ignite ships with a robust {cpp} thick client.
```
Comment (isapego) [inline docs/_docs/quick-start/cpp.adoc:41]: Not sure about these changes

**Review by isapego — APPROVED** [review]
Looks good in general

**Review by isapego — APPROVED** [review]

File: docs/_docs/SQL/ODBC/odbc-driver.adoc [hunk 2]
```diff
@@ -163,62 +163,57 @@ Apache Ignite ODBC Driver was officially tested on:
 
 == Building ODBC Driver
 
-Ignite is shipped with pre-built installers for both 32- and 64-bit versions of the driver for Windows. So if you just want to install ODBC driver on Windows you may go straight to the <<Installing ODBC Driver>> section for installation instructions.
+Ignite is shipped with pre-built installers for both 32- and 64-bit versions of the driver for Windows.
+So if you just want to install ODBC driver on Windows you may go straight to the <<Installing ODBC Driver>>
```
Comment (Nikita-tech-writer) [inline docs/_docs/SQL/ODBC/odbc-driver.adoc:167]:
```suggestion
So if you just want to install ODBC driver on Windows, you may go straight to the <<Installing ODBC Driver>>
```

File: docs/_docs/SQL/ODBC/odbc-driver.adoc [hunk 3]
```diff
@@ -163,62 +163,57 @@ Apache Ignite ODBC Driver was officially tested on:
...
+section for installation instructions.
 
-If you use Linux you will still need to build ODBC driver before you can install it. So if you are using Linux or if you still want to build the driver by yourself for Windows, then keep reading.
-
-Ignite ODBC Driver source code is shipped as part of the Ignite package and it should be built before usage.
-
-Since the ODBC Driver is written in {cpp}, it is shipped as part of Ignite {cpp} and depends on some of the {cpp} libraries. More specifically, it depends on the `utils` and `binary` Ignite libraries. This means that you will need to build them prior to building the ODBC driver itself.
-
-We assume here that you are using the binary Ignite release. If you are using the source release, instead of `%IGNITE_HOME%\platforms\cpp` path you should use `%IGNITE_HOME%\modules\platforms\cpp` throughout.
+If you use Linux you will still need to build ODBC driver before you can install it.
```
Comment (Nikita-tech-writer) [inline docs/_docs/SQL/ODBC/odbc-driver.adoc:170]:
```suggestion
For Linux, you will still need to build an ODBC driver before installing it.
```

File: docs/_docs/SQL/ODBC/odbc-driver.adoc [hunk 4]
```diff
@@ -163,62 +163,57 @@ Apache Ignite ODBC Driver was officially tested on:
...
+So if you are using Linux or if you still want to build the driver by yourself for Windows, then keep reading.
```
Comment (Nikita-tech-writer) [inline docs/_docs/SQL/ODBC/odbc-driver.adoc:171]:
```suggestion
So if you are using Linux or still want to build the driver yourself for Windows, refer to the next section.
```

File: docs/_docs/SQL/ODBC/odbc-driver.adoc [hunk 5]
```diff
@@ -163,62 +163,57 @@ Apache Ignite ODBC Driver was officially tested on:
...
 
-You will need MS Visual Studio 2010 or later to be able to build the ODBC driver on Windows. Once you have it, open Ignite solution `%IGNITE_HOME%\platforms\cpp\project\vs\ignite.sln` (or `ignite_86.sln` if you are running 32-bit platform), left-click on odbc project in the "Solution Explorer" and choose "Build". Visual Studio will automatically detect and build all the necessary dependencies.
-
-The path to the .sln file may vary depending on whether you're building from source files or binaries. If you don't see your .sln file in `%IGNITE_HOME%\platforms\cpp\project\vs\`, try looking in `%IGNITE_HOME%\modules\platforms\cpp\project\vs\`.
-
-NOTE: If you are using VS 2015 or later (MSVC 14.0 or later), you need to add `legacy_stdio_definitions.lib` as an additional library to odbc project linker's settings in order to be able to build the project. To add this library to the linker input in the IDE, open the context menu for the project node, choose `Properties`, then in the `Project Properties` dialog box, choose `Linker`, and edit the `Linker Input` to add `legacy_stdio_definitions.lib` to the semi-colon-separated list.
+You should install these dependencies:
```
Comment (Nikita-tech-writer) [inline docs/_docs/SQL/ODBC/odbc-driver.adoc:175]:
```suggestion
To start with, install the dependencies below:
```

File: docs/_docs/SQL/ODBC/odbc-driver.adoc [hunk 6]
```diff
@@ -163,62 +163,57 @@ Apache Ignite ODBC Driver was officially tested on:
...
 
-Once the build process is complete, you can find `ignite.odbc.dll` in `%IGNITE_HOME%\platforms\cpp\project\vs\x64\Release` for the 64-bit version and in `%IGNITE_HOME%\platforms\cpp\project\vs\Win32\Release` for the 32-bit version.
+* MS Visual C++ (10.0 and up), g++ (4.4.0 and up)
+* OpenSSL (32-bit and 64-bit versions)
```
Comment (Nikita-tech-writer) [inline docs/_docs/SQL/ODBC/odbc-driver.adoc:178]: (32-bit **or** 64-bit versions) ?

File: docs/_docs/SQL/ODBC/odbc-driver.adoc [hunk 7]
```diff
@@ -163,62 +163,57 @@ Apache Ignite ODBC Driver was officially tested on:
...
+* CMake 3.6+
+* http://wixtoolset.org[WiX Toolset] and add it to `%Path%`.
```
Comment (Nikita-tech-writer) [inline docs/_docs/SQL/ODBC/odbc-driver.adoc:180]:
```suggestion
* Install link:http://wixtoolset.org[WiX Toolset, window=_blank] and add it to `%Path%`.
```

File: docs/_docs/SQL/ODBC/odbc-driver.adoc [hunk 8]
```diff
@@ -163,62 +163,57 @@ Apache Ignite ODBC Driver was officially tested on:
...
 
-NOTE: Be sure to use the corresponding driver (32-bit or 64-bit) for your system.
-
-=== Building installers on Windows
-
-Once you have built driver binaries you may want to build installers for easier installation. Ignite uses link:http://wixtoolset.org[WiX Toolset] to generate ODBC installers, so to build them you'll need to download and install WiX. Make sure you have added the `bin` directory of the WiX Toolset to your PATH variable.
-
-Once everything is ready, open a terminal and navigate to the directory `%IGNITE_HOME%\platforms\cpp\odbc\install`. Execute the following commands one by one to build installers:
+Then do following:
```
Comment (Nikita-tech-writer) [inline docs/_docs/SQL/ODBC/odbc-driver.adoc:182]:
```suggestion
Then, perform the following:
```

File: docs/_docs/SQL/ODBC/odbc-driver.adoc [hunk 9]
```diff
@@ -163,62 +163,57 @@ Apache Ignite ODBC Driver was officially tested on:
...
 
+. Navigate to the `%IGNITE_HOME%\platforms\cpp` folder.
+. Build drivers and installers using following steps:
```
Comment (Nikita-tech-writer) [inline docs/_docs/SQL/ODBC/odbc-driver.adoc:185]:
```suggestion
. Build drivers and installers using the following steps:
```

File: docs/_docs/SQL/ODBC/odbc-driver.adoc [hunk 10]
```diff
@@ -163,62 +163,57 @@ Apache Ignite ODBC Driver was officially tested on:
...
-candle.exe ignite-odbc-x86.wxs
-light.exe -ext WixUIExtension ignite-odbc-x86.wixobj
+mkdir cmake-build-release-32
+cmake .. -DWITH_CORE=OFF -DWITH_ODBC=ON -DWITH_ODBC_MSI=ON -DCMAKE_GENERATOR_PLATFORM=Win32 -DOPENSSL_ROOT_DIR=<openssl 32-bit install dir> -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=..\install\x86
+cmake --build . --target install --config Release
 ----
 --
 
-As a result, `ignite-odbc-amd64.msi` and `ignite-odbc-x86.msi` files should appear in the directory. You can use them to install your freshly built drivers.
+As a result, `ignite-odbc-amd64.msi` and `ignite-odbc-x86.msi` files should appear
```
Comment (Nikita-tech-writer) [inline docs/_docs/SQL/ODBC/odbc-driver.adoc:206]:
```suggestion
As a result, `ignite-odbc-amd64.msi` and `ignite-odbc-x86.msi` files should appear.
```

**Pull request closed without merging**